packages/core/src/chart.ts

```diff
@@ -0,0 +1,193 @@
+// Internal Imports
+import { ChartConfig, BaseChartOptions, LayoutGrowth, LayoutDirection, LegendOrientations, ChartTheme } from "./interfaces/index";
+
+// Misc
+import { ChartModel } from "./model";
+import { Component, Title, Legend, LayoutComponent, Tooltip } from "./components";
+import { Tools } from "./tools";
+
+// Services
+import { DOMUtils, Events, Transitions } from "./services/index";
+import { select } from "d3-selection";
+
+export class Chart {
+	components: Array<Component>;
+	model: ChartModel = new ChartModel();
+	protected services: any = {
+		domUtils: DOMUtils,
+		events: Events,
+		transitions: Transitions
+	};
+
+	constructor(holder: Element, chartConfigs: ChartConfig<BaseChartOptions>) {}
+
+	// Contains the code that uses properties that are overridable by the super-class
+	init(holder: Element, chartConfigs: ChartConfig<BaseChartOptions>) {
+		// Set model update callback
+		this.model.setUpdateCallback(this.update.bind(this));
+
+		// Store the holder in the model
+		this.model.set({
+			holder
+		}, true);
+
+		// Set model data & options
+		this.model.setData(chartConfigs.data);
+
+		// Initialize all services
+		Object.keys(this.services).forEach(serviceName => {
+			const serviceObj = this.services[serviceName];
+			this.services[serviceName] = new serviceObj(this.model, this.services);
+		});
+
+		// Set chart resize event listener
+		this.services.events
+			.getDocumentFragment()
+			.addEventListener("chart-resize", () => {
+				this.update(false);
+			});
+
+		this.components = this.getComponents();
+
+		this.update();
+
+		this.setTheme();
```

Comment:
@natashadecoste 
`setTheme()` doesn't belong here.

We might need to add a `themes` service, and also add an `update()` method to services that `chart.ts`'s `update()` would call...